### PR TITLE
[MODNOTES-251,252]Update  to Java 17 and Spring boot v3.0.0 and identify issues.

### PR DESCRIPTION
**Pull request by KyryloShulzhenko**
## Links
https://issues.folio.org/browse/MODNOTES-251
https://issues.folio.org/browse/MODNOTES-252
## Purpose/Overview:
- Update the module to Spring boot v3.0.0 and identify issues.
- Update the module to Java 17 and identify issues.
- https://wiki.folio.org/display/TC/JDK+17+and+Java+17
## Acceptance criteria:
- Identify issues related to Java 17 update
- Identify issues related to Spring boot v3.0.0 update
- The module compiles without errors

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <parent>
```
Comment (psmagin) [inline pom.xml:5]: Please update module version to next major version

File: src/test/java/org/folio/notes/controller/NotesControllerTest.java [hunk 2]
```diff
@@ -817,7 +817,7 @@ void shouldReturn400WhenOrderByUpdatedDateParameterIsInvalid() throws Exception
   @Test
   @DisplayName("Should return list of notes searched by content case insensitively")
   void shouldReturnListOfNotesSearchedByContent() throws Exception {
-    var firstNote = generateNote().title("Title ABC").content("<p>test content</p><p>zztest</p>");
+    var firstNote = generateNote().title("Title ZZC").content("<p>test content</p><p>zztest</p>");
```
Comment (psmagin) [inline src/test/java/org/folio/notes/controller/NotesControllerTest.java:820]: Please do not change tests

File: src/main/java/org/folio/notes/domain/repository/NoteRepository.java [hunk 3]
```diff
@@ -39,11 +37,7 @@ static Specification<NoteEntity> domainEq(String domain) {
   }
 
   static Specification<NoteEntity> contentLike(String text) {
-    return (root, query, cb) -> new IlikePredicate(
-      (CriteriaBuilderImpl) cb,
-      root.get(NoteEntity_.indexedContent),
-      "%" + text + "%"
-    );
+    return (root, query, cb) -> cb.like(root.get(NoteEntity_.indexedContent), "%" + text + "%");
```
Comment (psmagin) [inline src/main/java/org/folio/notes/domain/repository/NoteRepository.java:40]: Please return the previous logic. It should be case-insensitive

**Review by psmagin — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/notes/domain/repository/NoteRepository.java (hunk 3 above)
Comment (KyryloShulzhenko) [reply src/main/java/org/folio/notes/domain/repository/NoteRepository.java]: Done

File: pom.xml (hunk 1 above)
Comment (KyryloShulzhenko) [reply pom.xml:5]: Done

**Review by psmagin — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Review by hasanepam — APPROVED** [review]

**Pull request merged**